### ISSUE TITLE
Web UI should provide information on client certificates created for workload clusters

### ISSUE THREAD

**Issue by snizhana-dynnyk** [giantswarm/roadmap#536]
## User story

- _As an admin, I want to see which client certificates have been created for workload cluster access, and could still be in use, so that I can adapt RBAC permissions if necessary._

## Background

Currently the web UI does not provide information on client certificates for workload cluster access created via the Management API (MAPI). Rest API users were used to getting that information.

The method to create certificates via the MAPI is deprecated (based on Giant Swarm's cert-operator and the CertConfig custom resource). We don't want to invest any effort into additional visibility for certificates created based on this method. Instead, we are looking for a solution that works for the Cluster API generation.

Access to the information should not only be available to the web UI, but also via kubectl.

TODO:

- [ ] Look into CSRs with cert-manager for client certificate creation

**Comment by marians** [comment]
> Duplicate of giantswarm/giantswarm#19161 ?

Yes

**Comment by PrashantR31** [comment]
These are the options ideas:

1. List in Happa so as to show all Workload Cluster client certificates and keep track of the key pairs that have been created and could be in use currently
2. Having metadata and exposing it to the Management API to that we can add it to Grafana and we can add needed alerting when/if needed

Both solutions need some more "Refinement time" to estimate the additional tasks needed.

Thoughts on UI/ Usability:

1. We already have tab for listing the key pairs in Happa, we can just the rest there, if the effort is low compared to the 2nd option.
2. Is the 2nd options be better suited for non-happa users ?

**Comment by marians** [comment]
We just had a customer ask about this

**Comment by PrashantR31** [comment]
I'm adding it to discuss further on this Friday (11Feb) Refinement.

**Comment by marians** [comment]
Team rainbow thinks this is currently blocked and we need talking with Phoenix and Rocket to find out details about client certs in CAPI. We'll invite to a call.

**Comment by marians** [comment]
Internal Slack thread: https://gigantic.slack.com/archives/CNX3Z835G/p1653041370862069

**Comment by anvddriesch** [comment]
We think that clientcerts should not be signed on the client side and we should reintroduce a way to keep track of the active certificates. Next step is a spike on how we want to do it.
We should ask cabbage what they think about using CSRs.

**Comment by marians** [comment]
Blocked by https://github.com/giantswarm/giantswarm/issues/15981 as we cannot display any details as long as we don't know whether/how we would actually create client certificates.

**Comment by weatherhog** [comment]
not possible. closing
